nanoc-core/lib/nanoc/core/utils.rb

```diff
@@ -14,13 +14,21 @@ module Utils
       def self.expand_path_without_drive_identifier(file_name, dir_string)
         res = File.expand_path(file_name, dir_string)
 
-        if Nanoc::Core.on_windows?
+        if windows_fs?
           # On Windows, strip the drive identifier, e.g. `C:`.
           res = res.sub(/^[A-Z]:/, '')
         end
 
         res
       end
+
+      # Returns `true` if absolute file paths start with a drive identifier, like `C:`.
+      def self.windows_fs?
+        @_windows_fs ||= begin
```

Comment:
On Mac / Linux this will usually return `false`, so for the memoization to work it should probably be something like `@_windows_fs = ...   if @_windows_fs.nil?`

Reply:
Oh! Good catch --- will fix.

(I have been writing an article on memoization in Ruby. The *irony*.)